src/proc/formats-converter.cpp

```diff
@@ -35,26 +67,32 @@ stream_profiles formats_converter::get_all_possible_profiles( const stream_profi
 
     for( auto & raw_profile : raw_profiles )
     {
-        for( auto && pbf : _pb_factories )
+        for( auto & pbf : _pb_factories )
         {
-            const auto && sources = pbf->get_source_info();
-            for( auto && source : sources )
+            const auto & sources = pbf->get_source_info();
+            for( auto & source : sources )
             {
                 if( source.format == raw_profile->get_format() &&
                    ( source.stream == raw_profile->get_stream_type() || source.stream == RS2_STREAM_ANY ) )
                 {
-                    const auto & targets = pbf->get_target_info();
-                    // targets are saved with format and type only. Updating fps and resolution before using as key
-                    for( auto target : targets )
+                    auto targets = pbf->get_target_info();
+                    // targets are saved with format, type and sometimes index. Updating fps and resolution before using as key
+                    for( auto & target : targets )
                     {
+                        // When interleaved streams are seperated to two distinct streams (e.g. sent as DDS streams),
+                        // same converters are registered for both stream. We handle the relevant one based on index.
+                        // Currently for infrared streams only.
+                        if( source.stream == RS2_STREAM_INFRARED && raw_profile->get_stream_index() != target.index )
```

Comment:
This is a bug fix?

Reply:
So it was an issue before, too? This is a bug fix?

Reply:
In the current USB D400 cameras this is not an issue but could be in other modules if the FW will stream two Y8 not one Y8I stream.

Reply:
Not sure I understand why it's not an issue with D400. If you think it's right, fine, I just wanted to understand if you're addressing an existing problem or a new one.